### DISCUSSION TITLE
Having project specific fixtures for each of the projects found in integrations folder

### DISCUSSION THREAD

**Discussion by iga-ob** [cypress-io/cypress#15427] [Questions and Help] [answered]
We are in the process of setting up the cypress testing infrastructure for several projects found under a common parent directory. One of them acts as the main directory (which we'll call the `main` project for this discussion) from which to run cypress, and is the only one meant to contain the `cypress.json` and `cypress.env.json` config files. However, we intend to also have the cypress folder structure for what the other projects might need, such as project specific plugins, fixtures, etc.

We have so far managed to import all of the tests by using a glob in the `testFiles` field, but we are still to find a way to place the fixtures in each of the projects which doesn't involve having to change the `main` project in some way. In other words, we would like to have some simple way in which we could import or make use of the fixtures in the tests by looking up the closest `fixtures` folder to them, and that is also compatible with executing the `npx cypress run` or `npx cypress open` commands to show and run all of the test specs without having to make use of some specific configuration file found at some arbitrary location.

We are currently researching the use of a plugin placed at the `main` project as a potential solution to this, but we would like to hear from any other simpler built-in solution or workaround that fits these needs. Thanks beforehand.

**Comment by bahmutov** [comment, marked as answer]
I believe the best solution would be either to:
- use separate config files via `--config-file <...>` option
- coding the logic to return the different fixture folder paths from the plugin file https://on.cypress.io/configuration#Overriding-Options

**Reply by iga-ob** [reply]
Hi bahmutov,

Thanks for your quick reply. After further research following your reply, I'm now attempting to do the `fixtures` path modification by identifying the test spec being run, via the `index.js` file found in the `main` project.

The contents of such file are as follows, based on cypress' documentation:

```
module.exports = (on, config) => {
  on('before:spec', spec => {
    console.log('Running a test: ', spec.relative);
  });

  console.log('RUNNING MY TEST SPEC');
};
```

But for some reason, only the "RUNNING MY TEST SPEC" log message appears when running `npx cypress open`. No message is printed when clicking on a test spec to run it. Why could this be?
